cilium/cmd/kernel_check.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"text/tabwriter"
+
+	"github.com/cilium/cilium/pkg/datapath/linux/probes"
+	"github.com/spf13/cobra"
+)
+
+var kernelCheckCmd = &cobra.Command{
+	Use:   "kernel-check",
+	Short: "Checks whether machine is supporting required kernel capabilities",
+	Run: func(cmd *cobra.Command, args []string) {
+		kernelCheck()
+	},
+}
+
+func kernelCheck() {
+	probeManager := probes.NewProbeManager()
+	w := tabwriter.NewWriter(os.Stdout, 30, 8, 0, '\t', 0)
+
+	requiredParams := probeManager.GetRequiredConfig()
+	fmt.Fprintf(w, "\n %s\t%s\t%s\t", "FEATURE:", "SUPPORTED:", "REQUIRED:")
```

Comment:
Won't this `\n` print a blank line above the column titles? Also I see you have `\n` both at the beginning and the end of your format strings in the following `Fprintf()`, won't that insert blank lines in the middle of the table?

Reply:
Well, yes. I thought this might increase readability but I might be wrong here ;-) As with the current formatting in place the output looks like this: 

```
$ vagrant@runtime1:~/go/src/github.com/cilium/cilium sudo cilium kernel-check

 FEATURE:			SUPPORTED:			REQUIRED:
 CONFIG_HAVE_EBPF_JIT		 true				 true

 CONFIG_BPF			 true				 true

 CONFIG_BPF_SYSCALL		 true				 true

 CONFIG_NET_SCH_INGRESS		 false				 true

 CONFIG_NET_CLS_BPF		 false				 true

 CONFIG_NET_CLS_ACT		 true				 true

 CONFIG_BPF_JIT			 true				 true

CONFIG_LWTUNNEL_BPF		 false				 false

CONFIG_BPF_EVENTS		 true				 false

CONFIG_CGROUP_BPF		 false				 false
```
Do you want me to change the formatting?

Reply:
I would err on the side of making this consistent with the behaviour of the other `cilium` CLI options. Specifically:

* Colon is not necessary for columns
* No space before the title or between each entry

Reply:
@joestringer valid point 🙂 , adapted formatting accordingly - output now looks like this (see below), what do you think? 

```
vagrant@runtime1:~/go/src/github.com/cilium/cilium$ sudo cilium kernel-check
FEATURE                       SUPPORTED                     REQUIRED
CONFIG_BPF                     true                          true
CONFIG_BPF_SYSCALL             true                          true
CONFIG_NET_SCH_INGRESS         false                         true
CONFIG_NET_CLS_BPF             false                         true
CONFIG_NET_CLS_ACT             true                          true
CONFIG_BPF_JIT                 true                          true
CONFIG_HAVE_EBPF_JIT           true                          true
CONFIG_CGROUP_BPF              false                         false
CONFIG_LWTUNNEL_BPF            false                         false
CONFIG_BPF_EVENTS              true                          false
```